### PR TITLE
build: remove 'ipykernel' dependency

### PR DESCRIPTION
**Pull request by marcofavoritobi**
## Proposed changes

The dependency `ipykernel` is not used directly from any module in `black-it`, hence it should not be declared as main dependency of the package. This PR removes it from the section `tool.poetry.dependencies` in `pyproject.toml`, and recomputes `poetry.lock` by using `poetry lock --no-update`.

Potentially breaking change if packages dependent of `black-it` use `ipykernel` directly (unlikely).

### CODE REVIEW

**Comment by muxator** [comment]
We should see if there are any examples/documentation that need to be updated

File: poetry.lock [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.6.1 and should not be changed by hand.
+# This file is automatically @generated by Poetry 1.5.1 and should not be changed by hand.
```
Comment (muxator) [inline poetry.lock:1]: Update your poetry version :)

Comment (marcofavoritobi) [reply poetry.lock:1]: :scream:

**Comment by marcofavoritobi** [comment]
> We should see if there are any examples/documentation that need to be updated

I don't think so, for the following reasons:

- `ipykernel` is never used directly, neither in docs nor in examples.
- documentation code and examples code rely on development dependencies, and `ipykernel` is included in them, due to e.g. `nbmake`, `jupyter` etc.
- `ipykernel` is needed by `ipywidgets`, which atm is a main dependency; therefore, `ipykernel` is always installed, hence nothing changed from the library perspective.

**Review by AldoGl — APPROVED** [review]

**Comment by muxator** [comment]
Closing in favor of #78, which includes this one.

**Pull request closed without merging**